packages/frontend/src/components/accounts/two_factor/TwoFactorVerifyModal.js

```diff
@@ -1,32 +1,125 @@
+import BN from 'bn.js';
+import { utils } from 'near-api-js';
 import React, { useEffect, useState } from 'react';
 import { Translate } from 'react-localize-redux';
 import { useDispatch, useSelector } from 'react-redux';
 import styled from 'styled-components';
 
 import { Mixpanel } from '../../../mixpanel/index';
-import { resendTwoFactor, get2faMethod } from '../../../redux/actions/account';
-import { selectAccountSlice } from '../../../redux/slices/account';
+import { resendTwoFactor, get2faMethod, getMultisigRequest } from '../../../redux/actions/account';
+import { selectAccountSlice, selectAccountMultisigRequest } from '../../../redux/slices/account';
 import { selectActionsPending, selectStatusSlice } from '../../../redux/slices/status';
 import { WalletError } from '../../../utils/walletError';
+import AlertBanner from '../../common/AlertBanner';
 import FormButton from '../../common/FormButton';
 import Modal from '../../common/modal/Modal';
 import ModalTheme from '../ledger/ModalTheme';
 import TwoFactorVerifyInput from './TwoFactorVerifyInput';
 
+const {
+    format: {
+        formatNearAmount
+    },
+    key_pair: {
+        PublicKey,
+        KeyType
+    },
+} = utils;
+
 const Form = styled.form`
     display: flex;
     flex-direction: column;
     align-items: center;
     width: 100%;
 `;
 
+const formatNear = (amount) => {
+    try {
+        return formatNearAmount(amount, 4);
+    } catch {
+        // Near amount may be stored as hex rather than decimal
+        return formatNearAmount(new BN(amount, 16).toString(), 4);
```

Comment:
No I wasn't able to figure out why 😢 . I _suspect_ it has something to do with `near-api-js` because when I query the contract directly with `near-cli` the values come back in the right format. This also seems specific to `functionCall` as I didn't encounter this with any of the other actions.

I'm planning to do some more digging to hopefully get to the bottom of this 😄 

```js
// from `near-api-js`
{
    "functionCall": {
        "methodName": "nft_buy",
        "args": [/* buffer */],
        "gas": "5af3107a4000",
        "deposit": "01a9e7b50fe5dae8400000"
    },
    "enum": "functionCall"
}
```
```js
// from `near-cli`
{
  actions: [
    {
      type: 'FunctionCall',
      method_name: 'nft_buy',
      args: 'eyJ0b2tlbl9zZXJpZXNfaWQiOiIyNjYiLCJyZWNlaXZlcl9pZCI6InNwdWQudGVzdG5ldCJ9',
      deposit: '2011280000000000000000000',
      gas: '100000000000000'
    }
  ]
}
```

Reply:
@MaximusHaximus do you have any idea why we'd see hex for `functionCall` from `near-api-js` ❓

Reply:
I'm still looking in to this, but for now i've opted to fetch and use the multisig request from the contract (`this.viewFunction()`) directly rather than request stored by `near-api-js` (`this.getRequest()`) as it doesn't store the Near values in hex.

https://github.com/near/near-wallet/pull/2523/files#diff-0592ce7ed5478b8cd8fd764e47824321d8a8bae9b77b2616da331efc69a02a94R51